rest/RESTClient.ts

```diff
@@ -63,11 +70,18 @@ export default class RESTClient {
         const discordBucket = api.headersIn['x-ratelimit-bucket'] as string | undefined;
         if (api.headersIn['x-ratelimit-global'] === 'true') {
           this.globallyRateLimited = true;
+
           setTimeout(() => {
             this.globallyRateLimited = false;
+
+            while (this.queue.length >= 1) {
+              this.queue.shift()?.();
+            }
           }, api.json.retry_after * 1e3);
+          // eslint-disable-next-line no-undefined
         } else if (discordBucket !== undefined) {
           const bucket = this.rateLimits.getBucket(discordBucket) || this.rateLimits.get(rateLimitRoute) as RESTBucket;
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
           bucket.bucket ?? (bucket.bucket = discordBucket);
```

Comment:
if you really want to use this syntax do something like:
```suggestion
          bucket.bucket = bucket.bucket ?? discordBucket;
```
Otherwise it's probably more logical to use a regular if

Reply:
Yeah there wouldn't be a difference if I just used an if statement as @smhman suggested. I switched it over to an if in the last commit there.